pointcloud_to_gridmap:
  ros__parameters:
    input_topic: /Sojourner/kinect_range/point_cloud # pointcloud2
    output_topic: /grid_map # gridmap
    map_frame_id: map
    center_frame_id: base_link
    publish_latency: true

    resolution: 0.15 # m/cell
    world_size:
      length: 30.0
      width: 30.0

    publish_rate: 1.0 # Hz

local_map_visualization:
  ros__parameters:
    grid_map_topic: /local_map
    grid_map_visualizations:
    - local_elevation_points
    - local_map_region
    - local_surface_normals
    - local_traversability_grid
    # - traversability_step_grid
    # - traversability_slope_grid
    local_surface_normals:
      type: vectors
      params:
        layer_prefix: normal_vectors_
        position_layer: elevation
        scale: 0.06
        line_width: 0.005
        color: 15600153 # red
    local_traversability_grid:
      type: occupancy_grid
      params:
        layer: traversability
        data_min: 0.0
        data_max: 1.0

    # local_traversability_step_grid:
    #   type: occupancy_grid
    #   params:
    #     layer: traversability_stepsize
    #     data_min: 0.0
    #     data_max: 1.0

    # local_traversability_slope_grid:
    #   type: occupancy_grid
    #   params:
    #     layer: traversability_slope
    #     data_min: 0.0
    #     data_max: 1.0

    local_elevation_points:
      type: point_cloud
      params:
        layer: elevation

    local_map_region:
      type: map_region
      params:
        color: 3289650
        line_width: 0.003

# global_map_visualization:
#   ros__parameters:
#     grid_map_topic: /grid_map
#     # grid_map_topic: /grid_map
#     # grid_map_visualizations: [elevation_points, map_region] 
#     grid_map_visualizations:
#     - map_elevation_points
#     - map_upper_bound_points
#     - map_lower_bound_points
#     map_elevation_points:
#       type: point_cloud
#       params:
#         layer: elevation
#     map_upper_bound_points:
#       type: point_cloud
#       params:
#         layer: upper_bound
#     map_lower_bound_points:
#       type: point_cloud
#       params:
#         layer: lower_bound
    

global_map_visualization:
  ros__parameters:
    grid_map_topic: /global_map
    # grid_map_topic: /grid_map
    # grid_map_visualizations: [elevation_points, map_region] 
    grid_map_visualizations:
    - global_elevation_points
    - global_map_region
    - global_surface_normals
    - global_traversability_grid
    # - traversability_step_grid
    # - traversability_slope_grid
    global_surface_normals:
      type: vectors
      params:
        layer_prefix: normal_vectors_
        position_layer: elevation
        scale: 0.06
        line_width: 0.005
        color: 15600153 # red
    global_traversability_grid:
      type: occupancy_grid
      params:
        layer: traversability
        data_min: 0.0
        data_max: 1.0

    global_elevation_points:
      type: point_cloud
      params:
        layer: elevation

    global_map_region:
      type: map_region
      params:
        color: 3289650
        line_width: 0.003
